lib/util.js

```diff
@@ -127,15 +127,19 @@ util.genRandomString = function(length) {
 };
 
 util.extend = function() {
+  // First object will be modified!
   var obj1 = arguments[0];
+  // Properties from other objects will be copied over
```

Comment:
Is it possible to use Object.assign() if it exists, and do this if it doesn't?

Reply:
I suppose we could. But our version is already bundled, and if Object.assign behaved differently then we might have a cross-browser bug. The way I've done this type of polyfill in the past is to detect browser server-side and then serve up poly(pony) fills alongside the app script.  We could also use babel-polyfill to generate a version for us, but it would still increase bundle size.

Reply:
At some point in the future we switch to Object.assign, remove util.extend, and tell our customers they must polyfill themselves to support older browsers. (React did this back in version 11 I think)